mlrun/api/utils/projects/follower.py

```diff
@@ -312,62 +330,72 @@ def _sync_projects(self, full_sync=False):
-        try:
-            db_projects = mlrun.api.crud.Projects().list_projects(
-                db_session, format_=mlrun.common.schemas.ProjectsFormat.name_only
+    def _store_projects_from_leader(self, db_session, db_projects, leader_projects):
+
+        db_projects_names = [project.metadata.name for project in db_projects.projects]
+
+        # Don't add projects in non-terminal state if they didn't exist before to prevent race conditions
+        filtered_projects = []
+        for leader_project in leader_projects:
+            if (
+                leader_project.status.state
+                not in mlrun.common.schemas.ProjectState.terminal_states()
+                and leader_project.metadata.name not in db_projects_names
+            ):
+                continue
+            filtered_projects.append(leader_project)
+
+        for project in filtered_projects:
+            # if a project was previously archived, it's state will be overriden by the leader
+            # and returned to normal here.
+            mlrun.api.crud.Projects().store_project(
+                db_session, project.metadata.name, project
+            )
+
+    def _archive_projects_missing_from_leader(
+        self, db_session, db_projects, leader_projects
+    ):
+        logger.info("Performing full sync")
+        leader_project_names = [project.metadata.name for project in leader_projects]
+        projects_to_archive = {
+            project.metadata.name: project for project in db_projects.projects
+        }
+        for project_name in leader_project_names:
+            if project_name in projects_to_archive:
+                del projects_to_archive[project_name]
+
+        for project_to_archive in projects_to_archive:
+            logger.info(
+                "Found project in the DB that is not in leader. Archiving...",
+                name=project_to_archive,
             )
-            # Don't add projects in non terminal state if they didn't exist before to prevent race conditions
-            filtered_projects = []
-            for leader_project in leader_projects:
-                if (
-                    leader_project.status.state
-                    not in mlrun.common.schemas.ProjectState.terminal_states()
-                    and leader_project.metadata.name not in db_projects.projects
-                ):
-                    continue
-                filtered_projects.append(leader_project)
-
-            for project in filtered_projects:
-                mlrun.api.crud.Projects().store_project(
-                    db_session, project.metadata.name, project
+            try:
+                projects_to_archive[
+                    project_to_archive
+                ].status.state = mlrun.common.schemas.ProjectState.archived
+                mlrun.api.crud.Projects().patch_project(
+                    db_session,
+                    project_to_archive,
+                    projects_to_archive[project_to_archive].dict(),
```

Comment:
Per @alonmr's comment previously - you could list just the names of projects, and here instead of patching the full dictionary you could just send the relevant change (i.e. `{status: {state: "archived"}}`) which should give you the expected results. 
This does need to be verified, though...